NREL/alfalfa
# Watch JS files in development docker compose

This functionality does slow down the initial load time of the docker-compose.dev.yml file due to having to webpack the contents after the server starts. Any touch of a JS file will trigger a rebuild, which could probably be sped up.

* add nodemon
* point worker_web multistage to dev
* Do not copy poetry lock and package-lock to containers
* when mounting web files, ignore build and node_modules

How to test:
* Launch the docker containers with `docker compose -f docker-compose.yml -f docker-compose.dev.yml up --build`
* run integration tests `poetry run pytest -m "integration"`
* touch one of the server JS files and make sure the site rebuilds
* monkey test the site as needed
